Testscripts/Windows/PRODUCTION-CHECKPOINT-PARTITION.ps1

```diff
@@ -135,27 +135,68 @@ function Main {
                 $availableDisks += $disk.Trim()
             }
         }
-        $mount_1st_disk = $availableDisks[0]
-        $mount_2nd_disk = $availableDisks[1]
-        Run-LinuxCmd -username $user -password $password -ip $Ipv4 -port $VMPort `
-            -command "mount /dev/${mount_1st_disk}1 /mnt/1; mount /dev/${mount_1st_disk}2 /mnt/2" -runAsSudo
+
+        # The disk swap, dual mount and resource disk make mount/umount complex
+        # It has to traverse all disks combination if DUALMOUNT is true
+        $disk_combination = @()
         if ($TestParams.DUALMOUNT) {
+            $rest = New-Object Collections.ArrayList
+            $availableDisks[1..($availableDisks.Length - 1)] | foreach { $rest.Add($_) | Out-Null }
+            $prefix = $availableDisks[0]
+            $(while ($rest) {
+                foreach ($suffix in $rest) {
+                    $disk_combination += $prefix
```

Comment:
$prefix is mostly static, why do you need tuple format of $disk_combination? I am concerned about duplication in the memory.

Reply:
For PRODUCTION-CHECKPOINT-IDE_SCSI test, the DUALMOUNT is true and extra two data disks are added. There are 3 data disks counting in resource disk.
Assume the below scenario:
- After running the PartitionDisks.sh, the $testfile1 is created on /dev/sdd.
- Swap between /dev/sdb and /dev/sdd after reboot the vm. 
    NOTE: The $testfile1 exists on /dev/sdb NOT /dev/sdd now.
- Because the DUALMOUNT is true, the "mount" sentence  will be executed twice, like this:
    -- mount /dev/sdb1 /mnt/1; mount /dev/sdb2 /mnt/2
    -- mount /dev/sdc1 /mnt/1; mount /dev/sdc2 /mnt/2
    NOTE: They mount the same point /mnt/1 and /mnt2, so the second mount will override the                    first one.
- Check-FileInLinuxGuest will return false based on the above conditions.
In my patch, it will try all combinations of disks, the below one will be executed:
    -- mount /dev/sdc1 /mnt/1; mount /dev/sdc2 /mnt/2
   -- mount /dev/sdb1 /mnt/1; mount /dev/sdb2 /mnt/2
- Check-FileInLinuxGuest will return true now.

Reply:
Above all, if $availableDisks=(sdb sdc sdd) , and the DUALMOUNT is true. The $disk_combination is equivalent to this:   ( (sdb sdc), (sdb,sdd), (sdc,sdd), (sdc,sdb), (sdd,sdb), (sdd,sdc)  )